src/lib/config/index.ts

```diff
@@ -329,9 +329,29 @@ export class ShellConfig {
         )}\n Resolve them by ensuring they have a secret defined or remove them if they are not needed.`,
         ...this.errors,
       ];
-    } else {
-      return [];
     }
+    if (!fileExistsWithPermission600(getRootConfigPath())) {
```

Comment:
this check will log a warning if the file exists and is non-readable.

imo this check should be the other way around, and check for `fileExistsWithNon600Permissions`, so that only one check is needed, and the log won't show up for non-readable files and such.